### PR TITLE
LP-2455 Add support for multilingual email template

### PR DESCRIPTION
**Pull request by javeriaejaz-arbisoft**
…erences

### CODE REVIEW

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/1319/workflows/workflows_id/jobs/1319/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/common/lib/mandrill_client/client.py (0.0%): Missing lines 74
openedx/adg/common/lib/mandrill_client/helpers.py (100%)
openedx/adg/common/lib/mandrill_client/tests/test_helpers.py (100%)
-------------
Total:   15 lines
Missing: 1 line
Coverage: 93%
-------------
```

**Review by MunamMubashir-arbisoft — APPROVED** [review]
LGTM!

**Review by ahmed-faraz46 — APPROVED** [review]

File: openedx/adg/common/lib/mandrill_client/tests/test_helpers.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+@pytest.mark.django_db
+def test_add_user_preferred_language_to_template_slug(user, slug, language, expected):
+    """
+    Assert that Mandrill function returns template slug as per user selected language
+    """
+    UserPreferenceFactory.build(
+        user=user,
+        key='pref-lang',
+        value=language,
+    ).save()
```
Comment (zahid-arbisoft) [inline openedx/adg/common/lib/mandrill_client/tests/test_helpers.py:32]: No need for a user fixture.

Code can be simplified to:

```suggestion
    UserPreferenceFactory(user=UserFactory(), key='pref-lang', value=language)
```

File: openedx/adg/common/lib/mandrill_client/tests/test_helpers.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    """
+    return UserFactory()
+
+
+@pytest.mark.parametrize('slug, language, expected', [
+    ('template-name', 'en', 'template-name'),
+    ('template-name', 'ar', 'template-name-ar')
+])
+@pytest.mark.django_db
+def test_add_user_preferred_language_to_template_slug(user, slug, language, expected):
```
Comment (zahid-arbisoft) [inline openedx/adg/common/lib/mandrill_client/tests/test_helpers.py:24]: Name suggestion:

`test_add_user_preferred_language_to_template_slug_successfully`

**Review by zahid-arbisoft — CHANGES_REQUESTED** [review]

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/1324/workflows/workflows_id/jobs/1324/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/common/lib/mandrill_client/client.py (50.0%): Missing lines 74
openedx/adg/common/lib/mandrill_client/helpers.py (100%)
openedx/adg/common/lib/mandrill_client/tests/test_helpers.py (100%)
-------------
Total:   20 lines
Missing: 1 line
Coverage: 95%
-------------
```

File: openedx/adg/common/lib/mandrill_client/tests/test_helpers.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    ('template-name', 'en', 'template-name'),
+    ('template-name', 'ar', 'template-name-ar')
+])
+@pytest.mark.django_db
+def test_add_user_preferred_language_to_template_slug_successfully(slug, language, expected):
+    """
+    Assert that Mandrill function returns template slug as per user selected language
+    """
+    user = UserFactory()
+    UserPreferenceFactory.build(user=user, key='pref-lang', value=language).save()
```
Comment (zahid-arbisoft) [inline openedx/adg/common/lib/mandrill_client/tests/test_helpers.py:21]: [build()](https://factoryboy.readthedocs.io/en/latest/#using-factories) returns user without saving it. Once we get an unsaved object then we are saving it manually.

`UserPreferenceFactory()` does both and is a shortcut. I think we should use it.

File: openedx/adg/common/lib/mandrill_client/helpers.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        string: template slug
+    """
+
+    user_pref = UserPreference.objects.filter(user__email=email, key='pref-lang').first()
+    user_lang = user_pref.value if user_pref else None
+
+    if user_lang and user_lang != settings.LANGUAGE_CODE:
+        return '{}-{}'.format(template, user_lang)
+
+    return template
```
Comment (asadali145) [inline openedx/adg/common/lib/mandrill_client/helpers.py:27]: We should use f-string and we can also improve this by

```suggestion
    return f'{template}-{user_lang}' if user_lang and user_lang != settings.LANGUAGE_CODE else template
```

**Review by zahid-arbisoft — APPROVED** [review]

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/1325/workflows/workflows_id/jobs/1325/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/common/lib/mandrill_client/client.py (50.0%): Missing lines 74
openedx/adg/common/lib/mandrill_client/helpers.py (100%)
openedx/adg/common/lib/mandrill_client/tests/test_helpers.py (100%)
-------------
Total:   20 lines
Missing: 1 line
Coverage: 95%
-------------
```

**Pull request merged**